kubespawner/spawner.py

```diff
@@ -36,12 +184,11 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # By now, all the traitlets have been set, so we can use them to compute
         # other attributes
-        self.executor = ThreadPoolExecutor(max_workers=self.k8s_api_threadpool_workers)
+        self.executor = SingletonExecutor.instance(max_workers=self.k8s_api_threadpool_workers)
+
+        # FIXME: Is this a memory leak? Will there be other side effects?
```

Comment:
Clarified comment - I was worried the thread that's doing the watching will live forever, but that's kinda what we wanna do anyway so it is ok.